From ac96548abbe8df86582f35e32986c852eb93c4d6 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 17 Sep 2024 10:35:14 +0200
Subject: [PATCH 1232/1695] dt-bindings: pinctrl: stm32mp: flags for non-secure
 pins

Define stm32 pinctrl DT bindings bit flags for pins that are
expected to be used in non-secure state.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git eb41a9e59670180fb4377ea958c08765e87f39a8 dt-bindings: pinctrl: stm32mp: flags for non-secure pins]
Change-Id: I120df1ce84f9c5fdabd94412b70e906cb905c7b1
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/404852
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/include/dt-bindings/pinctrl/stm32-pinfunc.h       |  5 +++++
 .../devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml  | 10 ++++++----
 2 files changed, 11 insertions(+), 4 deletions(-)

diff --git a/core/include/dt-bindings/pinctrl/stm32-pinfunc.h b/core/include/dt-bindings/pinctrl/stm32-pinfunc.h
index 370a25a93..afd118f8b 100644
--- a/core/include/dt-bindings/pinctrl/stm32-pinfunc.h
+++ b/core/include/dt-bindings/pinctrl/stm32-pinfunc.h
@@ -33,6 +33,11 @@
 
 #define STM32_PINMUX(port, line, mode) (((PIN_NO(port, line)) << 8) | (mode))
 
+#define STM32_PIN_NSEC		(1 << 31)
+
+#define STM32_PINMUX_NSEC(port, line, mode) \
+	(STM32_PIN_NSEC | STM32_PINMUX((port), (line), (mode)))
+
 /*  package information */
 #define STM32MP_PKG_AA	0x1
 #define STM32MP_PKG_AB	0x2
diff --git a/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml b/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
index 3be1efc83..dc9b9967d 100644
--- a/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
+++ b/documentation/devicetree/bindings/pinctrl/st,stm32-pinctrl.yaml
@@ -211,8 +211,10 @@ patternProperties:
               Integer array, represents gpio pin number and mux setting.
               Supported pin number and mux varies for different SoCs, and are
               defined in dt-bindings/pinctrl/<soc>-pinfunc.h directly.
-              These defines are calculated as: ((port * 16 + line) << 8) | function
+              These defines are calculated as:
+                (nsec-state << 31) | ((port * 16 + line) << 8) | function
               With:
+              - nsec-state: Hint on expected state of the pin: 0 if secure, 1 otherwise
               - port: The gpio port index (PA = 0, PB = 1, ..., PK = 11)
               - line: The line offset within the port (PA0 = 0, PA1 = 1, ..., PA15 = 15)
               - function: The function number, can be:
@@ -224,7 +226,7 @@ patternProperties:
               * 16 : Alternate Function 15
               * 17 : Analog
               * 18 : Reserved
-              To simplify the usage, macro is available to generate "pinmux" field.
+              To simplify the usage, macros are available to generate "pinmux" field.
               This macro is available here:
                 - include/dt-bindings/pinctrl/stm32-pinfunc.h
               Setting the pinmux's function to the Reserved (RSVD) value is used to inform
@@ -239,9 +241,9 @@ patternProperties:
                ... {
                           pinmux = <STM32_PINMUX('A', 9, GPIO)>;
                };
-               /* GPIO A9 set as analog */
+               /* Non-secure GPIO A9 set as analog */
                ... {
-                          pinmux = <STM32_PINMUX('A', 9, ANALOG)>;
+                          pinmux = <STM32_PINMUX_NSEC('A', 9, ANALOG)>;
                };
                /* GPIO A9 reserved for co-processor */
                ... {
-- 
2.47.2

